var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -91,38 +94,55 @@ def install(self, spec, prefix):
             options.append('--enable-openmp')
 
         if '+scalapack' in spec:
-            options.append('--with-scalapack=yes')
+            scalapack_option = 'intel' if '^intel-mkl' in spec else 'yes'
+            options.append('--with-scalapack={0}'.format(scalapack_option))
 
         if '+elpa' in spec:
-            if spec.satisfies('@:5.4.0'):
```

Comment:
oh, yes, i think that's exactly what will happen. I would keep it as-is in your PR currently and add a conflict statement to avoid using `@:5.4.0` with `elpa`.